PluginTemplateProject/src/main/scala/com/alpine/templates/db/TransformationTemplate.scala

```diff
@@ -0,0 +1,195 @@
+    //todo - set the output schema of the operatorSchemaManager to "outputSchema"
+    //NOTE - for now, output schema = input schema
+    if (inputSchemas.nonEmpty) {
+      val inputSchema = inputSchemas.values.iterator.next()
+      if (inputSchema.getDefinedColumns.nonEmpty) {
+        operatorSchemaManager.setOutputSchema(inputSchema)
+
+      }
+    }
+  }
+
+}
+
+/**
+ * The DBTransformationTemplateRuntime defines the runtime behavior of the operator.
+ */
+class DBTransformationTemplateRuntime extends DBRuntime[DBTable, DBTable] {
+
+  override def onExecution(context: DBExecutionContext,
+                           input: DBTable,
+                           params: OperatorParameters,
+                           listener: OperatorListener): DBTable = {
+
+    //get the basic database output parameters
+    val outputSchema = DBParameterUtils.getDBOutputSchemaParam(params)
+    val isView = DBParameterUtils.getIsViewParam(params)
+    val outputName = DBParameterUtils.getResultTableName(params)
+    val connectionInfo = context.getDBConnectionInfo
+    val overwrite = DBParameterUtils.getOverwriteParameterValue(params)
+    val fullOutputName = getQuotedSchemaTableName(outputSchema, outputName)
+    val tableOrView = if (isView) "VIEW" else "TABLE"
+
+    //check if there is a table  or with the same name as the output table and drop according to the
+    // "overwrite"
+    if (overwrite) {
+      val stmtTable = connectionInfo.connection.createStatement()
+      //First see if a table of that name exists.
+      // This will throw an exception if there is a view with the output name,
+      // we will catch the exception and delete the view in the next block of code.
+      try {
+        listener.notifyMessage("Dropping table if it exists")
+        val dropTableStatementBuilder = new StringBuilder()
+        dropTableStatementBuilder ++= "DROP TABLE IF EXISTS " + fullOutputName + " CASCADE;"
+        stmtTable.execute(dropTableStatementBuilder.toString())
+      }
+      catch {
+        case (e: Exception) => listener.notifyMessage("A view of the name " + fullOutputName + "exists")
+      }
+      finally {
+        stmtTable.close()
+      }
+      //Now see if there is a view with the output name
+      listener.notifyMessage("Dropping view if it exists")
+      val dropViewStatementBuilder = new StringBuilder()
+      dropViewStatementBuilder ++= "DROP VIEW IF EXISTS " + fullOutputName + " CASCADE;"
+      val stmtView = connectionInfo.connection.createStatement()
+      stmtView.execute(dropViewStatementBuilder.toString())
+      stmtView.close()
+    }
+
+    //todo - pull out values from custom parameters
+
+
+    //todo - create the sql statement that will be executed using the parameters
+    val sqlStatement = s"""
+      CREATE $tableOrView $fullOutputName AS (
+        SELECT *
+        FROM ${getQuotedSchemaTableName(input.schemaName, input.tableName)}
+      );
+
+        """
```

Comment:
But I think it would be okay to merge, and then add tests on a second iteration.